src/e2e/java/teammates/e2e/cases/AdminNotificationsPageE2ETest.java

```diff
@@ -0,0 +1,84 @@
+package teammates.e2e.cases;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.common.util.TimeHelper;
+import teammates.e2e.pageobjects.AdminNotificationsPage;
+
+/**
+ * SUT: {@link Const.WebPageURIs#ADMIN_NOTIFICATIONS_PAGE}.
+ */
+public class AdminNotificationsPageE2ETest extends BaseE2ETestCase {
+    private NotificationAttributes[] notifications = new NotificationAttributes[2];
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/AdminNotificationsPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+
+        notifications[0] = testData.notifications.get("notification1");
+        notifications[1] = testData.notifications.get("notification2");
+    }
+
+    @Test
+    @Override
+    public void testAll() {
+
+        ______TS("verify loaded data");
+        AppUrl url = createFrontendUrl(Const.WebPageURIs.ADMIN_NOTIFICATIONS_PAGE);
+        AdminNotificationsPage notificationsPage = loginAdminToPage(url, AdminNotificationsPage.class);
+        verifyPresentInDatabase(notifications[0]);
+        verifyPresentInDatabase(notifications[1]);
+        // Only validates that the notifications are present in the notifications table instead of checking every row
+        // This is because the page will display all notifications in the database, which is not predictable
+        notificationsPage.verifyNotificationsTableRow(notifications[0]);
+        notificationsPage.verifyNotificationsTableRow(notifications[1]);
+
+        ______TS("add new notification");
+        NotificationAttributes newNotification = NotificationAttributes
+                .builder("tm.e2e.ANotifications.notif1")
+                .withStartTime(TimeHelper.parseInstant("2035-04-01T22:00:00Z"))
+                .withEndTime(TimeHelper.parseInstant("2035-04-30T20:00:00Z"))
+                .withStyle(NotificationStyle.SUCCESS)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withTitle("E2E test notification 1")
+                .withMessage("<p>E2E test notification message</p>")
+                .build();
+
+        notificationsPage.addNotification(newNotification);
+        notificationsPage.verifyStatusMessage("Notification created successfully.");
+        String newestNotificationId = notificationsPage.getNewestNotificationId();
+        NotificationAttributes createdNotification = getNotification(newestNotificationId);
```

Comment:
We should still use `verifyPresentInDatabase` in place of `getNotification` here. While `verifyPresentInDatabase` eventually also calls `getNotification`, it performs retry in addition (in the case the data has yet to persist into DB it will retry a few times with delay in between). So `getNotification` cannot replace `verifyPresentInDatabase`. In general we want to verify the front end first (by matching the expected test input data with the actual data from the table) because once a user action is done, the front end usually gets updated immediately, but for the backend there's some delay before data gets persisted.

Reply:
Got it, I will remove `getNotification`! I will also remove `createdNotification` and just use `newNotification` for all the tests. However I think `verifyPresentInDatabase` will still have to be called before `verifyNotificationsTableRow` so that `createdAt` can be updated in `newNotification` before checking the table row values.

Reply:
`verifyNotificationsTableRow` only verifies the frontend data, there should be no need to involve data from DB I think

> However I think `verifyPresentInDatabase` will still have to be called before `verifyNotificationsTableRow` so that `createdAt` can be updated in `newNotification` before checking the table row values.

Yes you're right in this case we can't verify `createdAt` before the entity gets created, but we probably don't need to check the `createdAt` since it is auto-generated by the system.

Reply:
In this particular case, I think it is necessary for `createdAt` to be updated in `newNotification` first since the frontend displays the notification creation time, and therefore `createdAt` will be checked when calling `verifyTableRowValues`.